### ISSUE TITLE
[New source] Telegram channels

### ISSUE THREAD

**Issue by namazso** [ImranR98/Obtainium#1]
Modded apks are often distributed this way. Would be great to also have a regex to filter filenames to grab and install, as various builds are often just dumped into the same channel.

**Comment by ImranR98** [comment]
This can definitely be looked into. Could you provide a link to a Telegram channel that could be used for testing?

**Comment by namazso** [comment]
https://t.me/TwiFucker

provides modded twitter apks with adblocking

**Comment by ImranR98** [comment]
It seems like Telegram isn't a very open platform for app distribution as there's no way to get an APK without logging in with an account. The channel preview at https://t.me/s/TwiFucker does show the most recent messages with APK attachments, but clicking on one just prompts you to open the app.

We could ask the user to create a bot account, then use their account own name and API token to get access, but that is more complex than just adding a new source by extending `AppSource` - it would require Telegram-specific changes like saving user credentials, etc. Will leave the issue open but may not get around to it anytime soon.

**Comment by ImranR98** [comment]
Doesn't seem like this is likely to have any movement. Closing - will reopen if anything changes.

**Comment by ImranR98** [comment]
This issue has come up again recently and may be worth another look.

Since this issue was created, the AppSource class has been extended to allow for Source-specific additional options, so asking a user to add (non optional) API keys should be a lot easier.

**Comment by hg42** [comment]
I actually have quite a lot of apps from telegram channels. It happens to be the fastest medium for developers that already use telegram to chat with their users.
There are also apps that cannot be obtained anywhere else.
E.g. my own test releases of Neo Backup only appear in a group (! not even a channel).
I don't use other social media very often, but I guess there might be more hidden eco systems (discourse? reddit?).

**Comment by re7gog** [comment]
Probably it's possible by using https://pub.dev/packages/tdlib
Like:
1) user logins
2) selects chat or channel
3) then sets regex

But it's very difficult 🦭

**Comment by kaorlol** [comment]
> Probably it's possible by using https://pub.dev/packages/tdlib Like:
> 
>     1. user logins
> 
>     2. selects chat or channel
> 
>     3. then sets regex
> 
> 
> But it's very difficult 🦭

try it ig

**Comment by livxbox5** [comment]
How to add an application from a telegram channel and so that it finds new versions I can not add 
https://t.me/alexstranniklite
Приложением lazymedia deluxe


![Screenshot_2024-08-23-20-33-37-075_dev imranr obtainium fdroid](https://github.com/user-attachments/assets/421f8ae4-0130-4f3e-9ba3-55b0c1082cb6)
![Screenshot_2024-08-23-20-33-49-095_org telegram messenger](https://github.com/user-attachments/assets/c4277c2b-7b35-4ac2-b901-79d18edd2471)

**Comment by re7gog** [comment]
> How to add an application from a telegram channel and so that it finds new versions I can not add https://t.me/alexstranniklite Application lazymedia deluxe
> 
> ![Screenshot_2024-08-23-20-33-37-075_dev imranr obtainium fdroid](https://private-user-images.githubusercontent.com/76809274/361016380-421f8ae4-0130-4f3e-9ba3-55b0c1082cb6.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0OTc5NzksIm5iZiI6MTcyNDQ5NzY3OSwicGF0aCI6Ii83NjgwOTI3NC8zNjEwMTYzODAtNDIxZjhhZTQtMDEzMC00ZjNlLTliYTMtNTViMGMxMDgyY2I2LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI0VDExMDc1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2YmI2ZmQ5YTNjZmZkMDk5ZjI0YTYzOTZjM2QwNTQ2NmEzNjk5YTljODkyMGM5OTkxMTNkZmU2NjkxNTQ4NGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fOBwEC0dmbw3sWjP-oIVhhjQVD6ssymKZNoksIaw5C8) ![Screenshot_2024-08-23-20-33-49-095_org telegram messenger](https://private-user-images.githubusercontent.com/76809274/361016388-c4277c2b-7b35-4ac2-b901-79d18edd2471.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0OTc5NzksIm5iZiI6MTcyNDQ5NzY3OSwicGF0aCI6Ii83NjgwOTI3NC8zNjEwMTYzODgtYzQyNzdjMmItN2IzNS00YWMyLWI5MDEtNzlkMThlZGQyNDcxLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI0VDExMDc1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkwOWEyMzMwYmUwMDY4YzgwOWIyNmRkMTQ2NzQ2MzJhMDY5NWQ2NWU5MjRlYjdkY2Y5YTQ0MDY3YWU5YzlhNmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DMR7nNLFytk8jpiPij-Dm6D2SmE4EEuEnKk9InHQiAA)

чел, это не из телеграм каналов качать, а сам телеграм(да немного вводит в заблуждение)

**Comment by KScheid** [comment]
Why is telegram app listed as a source if you can't actually add it to obtanium?
![Screenshot_20241025_174128](https://github.com/user-attachments/assets/d168d614-165a-4dc5-8a35-89fa29fc570c)

**Comment by DwainZwerg** [comment]
No: This source is the Telegram app. Not an app in Telegram, but the Telegram app. However, this issue is about apps in Telegram, not the Telegram app.

**Comment by Pocarai** [comment]
### Is there an update for this?

I want keep in updated my modded app using this app

**Comment by DwainZwerg** [comment]
No. If you are an app dev (which I assume is the case when you issue a mod) you could work on the the source as re7gog described above:
> Probably it's possible by using https://pub.dev/packages/tdlib Like:
> 
>     1. user logins
> 
>     2. selects chat or channel
> 
>     3. then sets regex
> 
> 
> But it's very difficult 🦭

**Comment by aZuiZui** [comment]
This would be a great feature. Loads of app are distributed this way.

**Comment by Pocarai** [comment]
> No. If you are an app dev (which I assume is the case when you issue a mod) you could work on the the source as re7gog described above:
> 
> > Probably it's possible by using https://pub.dev/packages/tdlib Like:
> > ```
> > 1. user logins
> > 
> > 2. selects chat or channel
> > 
> > 3. then sets regex
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But it's challenging 🦭

I'm confused, and it isn't easy. Can I able to do that on Android

**Comment by rifatorigin** [comment]
Can Anyone show me step by step guide how to add a Telegram channel as a source for Obtainium?

**Comment by lokesh-sparrow** [comment]
I'm trying to add an app from RBMods to Obtainium, but it doesn't seem to work. The app isn't detected, and I can't add it successfully.

Has anyone managed to add an RBMods app (such as Telegram) to Obtainium? If so, could you share the correct method or any settings I need to change?

I'm using the latest version of Obtainium.

Any help would be greatly appreciated. 

Thanks! 🙏
